src/main/java/org/folio/circulation/domain/LoanAndRelatedRecords.java

```diff
@@ -51,6 +54,14 @@ public LoanAndRelatedRecords withTimeZone(DateTimeZone newTimeZone) {
     return new LoanAndRelatedRecords(loan, requestQueue, newTimeZone);
   }
 
+  public JsonObject asJson() {
+    JsonObject json = new JsonObject();
+    write(json, "loan", loan.asJsonWithUserAndProxy());
+    write(json, "requestQueue", JsonObject.mapFrom(requestQueue));
+    write(json, "timeZone", JsonObject.mapFrom(timeZone));
+    return json;
+  }
```

Comment:
Why does this class need to be converted to JSON?

Reply:
I used all this context to publish audit information. And I faced with issues when trying to create json with JsonObject.mapFrom() method. So here and below I just tried to convert all needed information to json object, encode it and send to mod-audit subscriber.

Reply:
These classes (and most of the context classes) aren't really intended to be used for serialization or external publication.

If these classes are to be used directly by the log publisher, I would personally prefer that the conversion was moved to a separate mapping class and used specifically for that purpose.

(There is some background to my reaction, in the early days of mod-circulation, it was convenient to use JSON representations within some of the classes and to do the mapping within those classes. As the system has grown far beyond an intermediary for storage, this design has caused significant issues and there is an intent to move this JSON processing to the outside edges of the module)